hathor/consensus/poa/poa_block_producer.py

```diff
@@ -126,11 +132,15 @@ def _schedule_block(self) -> None:
             return
 
         self._last_seen_best_block = previous_block
+        signer_index = self._get_signer_index(previous_block)
+        if signer_index is None:
+            return
+
         now = self._reactor.seconds()
-        expected_timestamp = self._expected_block_timestamp(previous_block)
+        expected_timestamp = self._expected_block_timestamp(previous_block, signer_index)
         propagation_delay = 0 if expected_timestamp < now else expected_timestamp - now
 
-        self._reactor.callLater(propagation_delay, self._produce_block, previous_block)
+        self._delayed_call = self._reactor.callLater(propagation_delay, self._produce_block, previous_block)
```

Comment:
You might cancel this call if a new block is received at the expected height.

Reply:
Done in a separate PR: https://github.com/HathorNetwork/hathor-core/pull/1080

Reply:
Thanks!